### PR TITLE
Add a optional tracer implementation loader.

### PR DESCRIPTION
**Pull request by wu-sheng**
1. Sync the OT version up-to-date, 0.20.4
2. Sync the gprc version up-to-date, 1.0.2
3. Add a TracerLoader to get tracer instance by spi. It just a optional choose.

### CODE REVIEW

**Comment by wu-sheng** [comment]
@bensigelman , I can't choose a reviewer, just @ you instead.

**Comment by wu-sheng** [comment]
I signed it!

File: java/src/main/java/io/opentracing/contrib/TracerLoader.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import io.opentracing.NoopTracerFactory;
+import io.opentracing.Tracer;
+
+import java.util.Iterator;
+import java.util.ServiceLoader;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Created by wusheng on 2016/12/21.
```
Comment (bhs) [inline java/src/main/java/io/opentracing/contrib/TracerLoader.java:21]: Let's make a good comment here showing how this fits in, and let's also add something to the README.

File: java/src/main/java/io/opentracing/contrib/ServerTracingInterceptor.java [hunk 2]
```diff
@@ -35,7 +35,14 @@
     private final OperationNameConstructor operationNameConstructor;
     private final boolean streaming;
     private final boolean verbose;
-    private final Set<ServerRequestAttribute> tracedAttributes; 
+    private final Set<ServerRequestAttribute> tracedAttributes;
+
+    /**
+     * use spi mechanism to get tracer implementation.
```
Comment (bhs) [inline java/src/main/java/io/opentracing/contrib/ServerTracingInterceptor.java:41]: better comment here; perhaps `@see` the TracerLoader.

File: java/build.gradle [hunk 3]
```diff
@@ -47,19 +47,21 @@ repositories {
...
 dependencies {
-	compile 'io.grpc:grpc-core:1.0.0-pre1'
-	compile 'io.opentracing:opentracing-api:0.14.0'
-	testCompile 'io.opentracing:opentracing-mock:0.14.0'
-	testCompile 'io.grpc:grpc-protobuf:1.0.0-pre1'
-	testCompile 'io.grpc:grpc-netty:1.0.0-pre1'
-	testCompile 'io.grpc:grpc-stub:1.0.0-pre1'
+	compile 'io.grpc:grpc-core:1.0.2'
+	compile 'io.opentracing:opentracing-api:0.20.4'
+	compile 'io.opentracing:opentracing-noop:0.20.4'
```
Comment (bhs) [inline java/build.gradle:56]: good, thanks.

**Comment by wu-sheng** [comment]
@bensigelman , updated. But the comments on README are very easy, just mention the spi mechanism. I think maybe it's enough, because it's just optional.

**Comment by wu-sheng** [comment]
@bensigelman , is there anything else to change?

**Pull request closed without merging**